nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/status-history/status-history-chart/status-history-chart.component.ts

```diff
@@ -0,0 +1,680 @@
+            ])
+            .on('brush', brushed);
+
+        // context area
+        const brushNode: any = control.append('g').attr('class', 'brush').on('click', brushed).call(brush);
+
+        if (!!this.brushSelection) {
+            brush = brush.move(brushNode, this.brushSelection);
+        }
+
+        // add expansion to the extent
+        control
+            .select('rect.selection')
+            .attr('style', 'pointer-events: all;')
+            .on('dblclick', () => {
+                if (this.brushSelection !== null) {
+                    // get the y range (this value does not change from the original y domain)
+                    const yRange = yControl.range();
+
+                    // expand the extent vertically
+                    brush.move(brushNode, [
+                        [this.brushSelection[0][0], yRange[1]],
+                        [this.brushSelection[1][0], yRange[0]]
+                    ]);
+                }
+            });
+
+        // identify all nodes and sort
+        this.nodes = statusData
+            .filter((status) => {
+                return status.id !== NIFI_NODE_CONFIG.nifiInstanceId;
+            })
+            .sort((a: any, b: any) => {
+                return a.label < b.label ? -1 : a.label > b.label ? 1 : 0;
+            });
+
+        brushed();
+    }
+
+    private getChartMinHeight() {
+        const chartContainer = document.getElementById('status-history-chart-container')!;
+        const controlContainer = document.getElementById('status-history-chart-control-container')!;
+
+        const marginTop: any = controlContainer.computedStyleMap().get('margin-top');
+        return (
+            chartContainer.parentElement!.clientHeight - controlContainer.clientHeight - parseInt(marginTop.value, 10)
+        );
+    }
+
+    private getChartMaxHeight() {
+        const chartContainer = document.getElementById('status-history-chart-container')!;
+        const controlContainer = document.getElementById('status-history-chart-control-container')!;
+
+        const marginTop: any = controlContainer.computedStyleMap().get('margin-top');
+        const statusHistory = document.getElementsByClassName('status-history')![0];
+        const dialogContent = statusHistory.getElementsByClassName('dialog-content')![0];
+        const dialogStyles: any = dialogContent.computedStyleMap();
+        const bodyHeight = document.body.getBoundingClientRect().height;
+
+        return (
+            bodyHeight -
+            controlContainer.clientHeight -
+            50 -
+            parseInt(marginTop.value, 10) -
+            parseInt(dialogStyles.get('top')?.value) -
+            parseInt(dialogStyles.get('bottom')?.value)
+        );
+    }
+
+    private getChartMaxWidth() {
+        const chartContainer = document.getElementById('status-history-chart-container')!;
+        const controlContainer = document.getElementById('status-history-chart-control-container')!;
```

Comment:
`chartContainer` and `controlContainer` appear to be unused.

Reply:
actually, the entire method is not called. will remove.